### DISCUSSION TITLE
<help>

### DISCUSSION THREAD

**Discussion by Mamadou-oury** [community/community#48088] [Programming Help]
### Select Topic Area

Product Feedback

### Body

how to fixe this error please?

Command 'g++' not found, but can be installed with:
sudo apt install g++

**Comment by Fo00oX** [comment]
Install the suite of development tools first. Then go back to compile the software.

`yum groupinstall 'Development Tools'`

You could need much more than just the compiler. The Development Tools package includes the core development tools like automake, gcc, perl, python, flex, make, gdb, bison, and many more. To list all of the software in the package group, use yum as follows.

`yum group info 'Development Tools'`

For Fedora 20 (at least), you'll additionally need to install `gcc-c++.`

For Debian-based systems, install the suite of development tools as follows.

`apt-get install build-essential`

https://unix.stackexchange.com/questions/140350/linux-g-command-not-found

**Comment by cehfisher** [comment]
Hi @Mamadou-oury - It looks like your issue was mistakenly added to the accessibility discussion board. @queenofcorgis could you please move this to the correct category? Thanks!

**Comment by mranv** [comment]
The error message "Command 'g++' not found, but can be installed with: sudo apt install g++" indicates that the g++ compiler for C++ is not installed on your system. To resolve this issue, you can follow the instructions provided in the error message and install g++ using the package manager.

If you're using Ubuntu or any Debian-based Linux distribution, you can install g++ by running the following command in your terminal:

```
sudo apt install g++
```

This command will prompt you for your user password. After entering the password, the package manager will install the g++ compiler along with any necessary dependencies.

Once the installation is complete, you should be able to use the g++ command to compile your C++ programs.

If you're using a different Linux distribution, the package manager and the package name for g++ may be different. For example, on Fedora or CentOS, you can install g++ using the following command:

```
sudo dnf install gcc-c++
```

On Arch Linux and its derivatives, you can use:

```
sudo pacman -S gcc
```

In case you're using a different operating system, like macOS or Windows, the installation process may vary. For macOS, you can install g++ as part of the Xcode Command Line Tools, and for Windows, you can use an integrated development environment (IDE) like Visual Studio, which includes the necessary compilers.

After successfully installing g++, you should be able to compile and run your C++ programs without encountering the "Command 'g++' not found" error.

**Comment by queenofcorgis** [comment]
@mranv 
We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.
